integration_tests/e2e/refer/programmeHistory.cy.ts

```diff
@@ -319,6 +320,70 @@ context('Programme history', () => {
         programmeHistoryDetailsPage.shouldContainSourceTextArea()
         programmeHistoryDetailsPage.shouldContainButton('Continue')
       })
+
+      it('updates the entry and redirects to the programme history page', () => {
+        const updatedCourseParticipation = courseParticipationFactory.build({
+          ...courseParticipation,
+          outcome: {
+            detail: 'Some outcome details',
+            status: 'complete',
+            yearCompleted: 2023,
+          },
+          setting: {
+            location: 'A community location',
+            type: 'community',
+          },
+          source: 'The source',
+        })
+
+        const programmeHistoryDetailsPage = Page.verifyOnPage(ProgrammeHistoryDetailsPage)
+        programmeHistoryDetailsPage.selectSetting(updatedCourseParticipation.setting.type as string)
+        programmeHistoryDetailsPage.inputSettingLocation(updatedCourseParticipation.setting.location as string)
+        programmeHistoryDetailsPage.selectOutcome(updatedCourseParticipation.outcome.status as string)
+        programmeHistoryDetailsPage.inputYearCompleted(
+          (updatedCourseParticipation.outcome.yearCompleted as number).toString(),
+        )
+        programmeHistoryDetailsPage.inputOutcomeDetail(updatedCourseParticipation.outcome.detail as string)
+        programmeHistoryDetailsPage.inputSource(updatedCourseParticipation.source as string)
```

Comment:
Would explicitly typing `updatedCourseParticipation` as a `CourseParticipation` remove the need for all this casting?

Reply:
Tweaked this.